test/test_flat_shallow_water_model.jl

```diff
@@ -0,0 +1,42 @@
+using Oceananigans 
+using Oceananigans.Models
+using Profile
+using Oceananigans.Grids: halo_size
+
+Nx = 4
+
+topologies = (  (Periodic, Periodic, Bounded), 
+                (Periodic, Periodic, Flat),
+                (Periodic, Flat,     Flat),    
+                (Flat,     Flat,     Flat)
+                )
+sizes = ( (Nx,Nx,1), 
+          (Nx, Nx), 
+          (Nx), 
+          () 
+          )
+extents = ( (1, 1, 1), 
+            (1, 1), 
+            (1), 
+            ()
+            ) 
+
+halos = ( (3, 3, 3),
+          (3, 3),
+          (3),
+          ())
+
+for (iter, topo) in enumerate(topologies)
```

Comment:
We need to test all grids too.

Reply:
Right, but presently shallow water can only use `RegularRectilinearGrid`.  This will become more presseing when we play with other models.

Reply:
We can test `RegularLatitudeLongitudeGrid` with `HydrostaticFreeSurfaceModel`. It would be quite useful to be able to flatten the vertical dimension there.

Reply:
Ah, I didn't think about that.  Even though the equations won't be right (no metric terms) it should allow us to test to see if `Flat` works or not.  Will try that right now even

Reply:
Okay, I tried running `RegularLatitudeLongitudeGrid` in `ShallowWaterModel` and found that this is not as easy as I might have hoped.  First, it doesn't recognize `topology` and it needs `longitude, latitude, z` all to be defined.  

I am tempted to create another issue asking that we make it easier to switch between `RegularLatitudeLongitudeGrid` and `RegularRectilinearGrid`, which would make the testing in any model much easier.    At the moment we can't specify the topology so there is no way to choose `Flat` in the context of that grid.  Or am I missing something?

Reply:
What about `VerticallyStretchedRectilinearGrid` with `IncompressibleModel` and `Flat` in x- or y- direction?

Reply:
I am going to save this for a future PR as it is more involved and will need more attention